### PR TITLE
[sig-scale] add script and configuration files to automate the execution of performance test in our CI/CD system

### PR DESCRIPTION
**Pull request by marceloamaral**
Add configuration to run density scaling performance testing by varying the number of VMIs creation from 400, 600, up tp 800.

This configuration will be used for the Medium-scale performance test that will run daily on the performance cluster on our CI/CD system. This test is part of the planned performance tests being introduced to monitor performance regressions.

This PR is related to #6309

Signed-off-by: Marcelo Amaral <marcelo.amaral1@ibm.com>

/cc @fgimenez @rmohr @davidvossel @rthallisey 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by rthallisey** [comment]
What's you plan with the 100 VMI density test?  It's a func test currently, so how should we integrate the two load generation paths?

**Comment by marceloamaral** [comment]
This PR is related to #7083 as `hack/perfscale-tests.sh` contains the system warm-up solution before running the performance test, so it can collect all events from the Prometheus metric `rest_client_requests_total`.

**Comment by rthallisey** [comment]
>What's you plan with the 100 VMI density test? It's a func test currently, so how should we integrate the two load generation paths?
>
What I'm asking is why is our testing using two code paths: 1) https://github.com/kubevirt/kubevirt/blob/main/tests/performance/density.go#L67 and 2) tools/perfscale-load-generator/examples/workload/kubevirt-density/kubevirt-density-scale.yaml?  Probably makes sense to pick one now that we're expanding the test matrix.

**Comment by rthallisey** [comment]
>What I'm asking is why is our testing using two code paths: 
>
Never mind let's sort this out later.  Is the code here ready? Or are you still testing?

**Comment by marceloamaral** [comment]
@rthallisey sorry for the delay in responding here.
So I set the plan with the test some time ago: https://github.com/kubevirt/kubevirt/issues/6309

Functional testing was a good exercise to start a test, but as performance and functional tests are fundamentally
different, they should have separate implementation. Also, functional tests are not made to scale.
So the aim is to replace load generation with functional tests for `perfscale-load-generator`, and maybe eventually we can use another tool like `kube-burner`.

**Comment by marceloamaral** [comment]
/cc @dhiller

The PR [project-infra/pull/1838](https://github.com/kubevirt/project-infra/pull/1838) depends on the scripts introduced by this PR here.

So, could you review this PR as well?

/retest

**Comment by dhiller** [comment]
@marceloamaral you might need to run `make generate` again.

File: automation/perfscale-test.sh [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      echo "Dump kubevirt state:"
+      make dump
+      exit 1
+    fi
+  done
+  kubectl get pods -n $i
+done
+
+# build perfscale tools
+# make bazel-build
```
Comment (dhiller) [inline automation/perfscale-test.sh:81]: I have seen calls to `perfscale-load-generator` and `perfscale-audit` below, so isn't calling the bazel build required to use it?

**Review by dhiller — COMMENTED** [review]
One question inside. Besides that: were you able to run it locally already i.e. using phaino?

File: automation/perfscale-test.sh (hunk 1 above)
Comment (marceloamaral) [reply automation/perfscale-test.sh:81]: Thanks!
It was commented out by mistake during testing...

**Review by dhiller — APPROVED** [review]
/approve
/hold

Remove hold when you see fit, maybe others want to take a final look.

File: hack/perfscale-tests.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    sleep 30
+    export PERFSCALE_WORKLOAD=$_perfscale_workload_tmp
+fi
+
+start_timestamp=$(date -u +%Y-%m-%dT%TZ)
+# run the test
+perftest ${additional_test_args} ${FUNC_TEST_ARGS}
+
+# run audit tool to dump metrics
+if [[ ${PERFAUDIT} == "true" || ${PERFAUDIT} == "True" ]]; then
```
Comment (rthallisey) [inline hack/perfscale-tests.sh:79]: metrics get dumped by calling `perftest`.  Do we still need this?

Comment (marceloamaral) [reply hack/perfscale-tests.sh:79]: This is true for the `functional test`, but here we are using the `perfscale-load-generator` tool... So if we want to collect metrics from the audit tool, we need to call it after the test.

**Pull request merged**